### ISSUE TITLE
ForceField from Unparsable String Raises Wrong Exception

### ISSUE THREAD

**Issue by SimonBoothroyd** [openforcefield/openforcefield#629]
**Describe the bug**
Creating a `ForceField` object from an unparsable string yields an (I think) unexpected `TypeError` exception:

```
        # Parse content depending on type
        for parameter_io_format in io_formats_to_try:
            parameter_io_handler = self.get_parameter_io_handler(parameter_io_format)
    
            # Try parsing as a forcefield file or file-like object
            try:
                smirnoff_data = parameter_io_handler.parse_file(source)
                return smirnoff_data
            except ParseError as e:
                exception_msg = str(e)
            except (FileNotFoundError, OSError):
                # If this is not a file path or a file handle, attempt parsing as a string.
                try:
                    smirnoff_data = parameter_io_handler.parse_string(source)
                    return smirnoff_data
                except ParseError as e:
>                   exception_msg = str(e)
E                   TypeError: __str__ returned non-string (type ExpatError)

/xxx/lib/python3.7/site-packages/openforcefield/typing/engines/smirnoff/forcefield.py:959: TypeError
```

It seems that this line should instead just be formatting the error and then later an `IOError` should be raised (line 970).

**To Reproduce**

```
from openforcefield.typing.engines.smirnoff.forcefield import ForceField
force_field = ForceField("1234")
```

**Computing environment (please complete the following information):**
 - Operating system: MacOS X 10.15.5

The above was generated using 0.6.0 but this can also be reproduced in 0.7.0.

**Comment by mattwthompson** [comment]
Works the same on my machine, but I'm a little confused as to why? The overriden `MessageException.__str__` just returns `.msg` and it behaves as one would expect when called directly:

```
In [5]: from openforcefield.typing.engines.smirnoff.forcefield import ParseError

In [6]: str(ParseError('this is the message'))
Out[6]: 'this is the message'

In [7]: from openforcefield.utils.utils import MessageException

In [8]: str(MessageException('this is the other message'))
Out[8]: 'this is the other message'
```
Is there something about `str(obj)` vs. `obj.__str__()` I'm overlooking?

**Comment by SimonBoothroyd** [comment]
It seems the issue is in `io.py` line 207. The custom `ParseError` class is being initialised with exception directly rather than a stringified exception.

**Comment by mattwthompson** [comment]
Fixed by #630
